src/backup_sram.cpp

```diff
@@ -0,0 +1,69 @@
+#include <cstring> //For memcpy
+#include "stm32f4xx_pwr.h"
+#include "stm32f4xx_rcc.h"
+#include "backup_sram.h"
+
+void backup_sram_init()
+{
+    //The checksum function depends on the BackupData struct being a multiple of 32 bits
+    //If you change the BackupData struct, and it trips on this static_assert,
+    //either make the BackupData struct a multiple of 32 bits, or change the checksum method
+    static_assert(sizeof(rosflight_firmware::BackupData)%4==0,"Error with debug_info_t: Size is not multiple of 32 bits");
+#pragma GCC diagnostic push //Ignore old style cast from included library
```

Comment:
I would not be opposed to just wrapping this whole thing in an `ignored "-Wold-style-cast` and `ignored "-Wmissing-filed-initializers`.  As it currently stands, I think it's a little clunky with it all over the place.